pkg/api/types.go

```diff
@@ -2141,6 +2141,22 @@ const (
 
 	// SSHAuthPrivateKey is the key of the required SSH private key for SecretTypeSSHAuth secrets
 	SSHAuthPrivateKey = "ssh-privatekey"
+
+	// SecretTypeTLS contains information about SSL termination used with the Ingress resource.
+	//
+	// Required fields:
+	// - Secret.Data["key"] - TLS private key.
+	//   Secret.Data["cert"] - TLS certificate.
+	//   Secret.Data["chain"] - optional certificate chain.
+	// TODO: Consider supporting different formats, specifying CA/destinationCA.
+	SecretTypeTLS SecretType = "kubernetes.io/tls"
+
+	// TLSCertKey is the key for tls certificates in a TLS secert.
+	TLSCertKey = "cert"
+	// TLSPrivateKeyKey is the key for the private key field in a TLS secret.
+	TLSPrivateKeyKey = "key"
+	// TLSChainKey is the key for the certificate chain in a TLS secret.
+	TLSChain = "chain"
```

Comment:
I would have expected Key, Cert, and CA, not Chain.  Is chain distinct from the CA trust intermediaries?  @liggitt for someone who spends much more time in the common codebase for TLS for consistency in naming and exposition.